Implement Event Reminder Feature

farellano0 opened the pull request:
## Description

This pull request introduces a feature that enables users to set a specific date and time for each event to receive an automatic email reminder.

### Changes Made

- Integrated functionality for users to set reminders on events.
- Implemented scheduled email reminders using Action Mailer and Active Job.
- Utilized Sidekiq for background processing of email reminder scheduling and sending.

### Purpose

Enhance user experience by providing a convenient way for users to receive reminders for their events, improving event management and attendance.

## Additional Information

- **Screenshots:**
![image](https://github.com/BrightCoders-Institute/s5a1-administrador-de-eventos-team-03-s5a1-administrador-de-eventos/assets/73454963/92bdc662-782b-4df0-9653-19b03d49e2d7)

farellano0 committed:
Implemented Event Reminders

Co-authored-by: Zama <alexavdstiles@gmail.com>
Co-authored-by: etormasa <hector.meza.141@gmail.com>
Co-authored-by: AlejandroEscobarCampos <ae339640@gmail.com>
Co-authored-by: Alan Hernández <ahernandez102@ucol.mx>
Co-authored-by: juperameza <juperameza@hotmail.com>
Co-authored-by: Erik Gomez <erikgomez198755@outlook.com>

juperameza commented on app/views/bright_events/_filters.html.erb:
```diff
@@ -24,7 +24,7 @@
 
   <script>
     function updateMinDate() {
-      var dateFromValue = document.getElementById('date_from').value;
+      let dateFromValue = document.getElementById('date_from').value;
       document.getElementById('date_to').min = dateFromValue;
     }
   </script>
```
This isn't the way to do J's in rails!!!!

farellano0 replied:
Sorry, we will correct it as soon as possible.

MauricioNC commented on app/jobs/event_reminder_job.rb:
```diff
@@ -0,0 +1,11 @@
+  queue_as :default
+
+  def perform(event_id, event_updated_at)
+    # Do something later
```
Remove base code

farellano0 replied:
Right away!

JuanequeX commented on app/views/bright_events/_filters.html.erb:
```diff
@@ -24,7 +24,7 @@
 
   <script>
     function updateMinDate() {
-      var dateFromValue = document.getElementById('date_from').value;
+      let dateFromValue = document.getElementById('date_from').value;
       document.getElementById('date_to').min = dateFromValue;
     }
   </script>
```
Move this code to a JavaScript controller using Stimulus. It's not a good practice to include logic in views; separating Ruby or JavaScript logic from views improves maintainability.

farellano0 replied:
Done!

farellano0 committed:
Corrections of pull request

Using stimulus for use javascript on html.erb files.
Setting stimulus conections for be implemented.

Co-authored-by: Zama <alexavdstiles@gmail.com>
Co-authored-by: etormasa <hector.meza.141@gmail.com>
Co-authored-by: AlejandroEscobarCampos <ae339640@gmail.com>
Co-authored-by: juperameza <juperameza@hotmail.com>
Co-authored-by: angel23v <angelvazquezvirgen@gmail.com>
